### PR TITLE
Fix delete memory leak, jshint cleanup

### PR DESCRIPTION
**Pull request by nijikokun**
Fixes memory leak by using `delete` keyword to remove properties from an object, thus turning it into a _slow object_ and causing memory gain over time instead of being GC'd properly, instead we convert to a new object, avoiding lingering references and memory leak.

Other fixes are Jshint / string consistency, as well some grammatical fixes.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -60,9 +58,8 @@ function gravatar(settings) {
  * @return {String} hashed `word`
  * @api private
  */
-
-function md5(word) {
-  return crypto.createHash('md5').update(word).digest("hex");
+function md5 (word) {
```
Comment (cristiandouce) [inline index.js:64]: Missing newline.

File: index.js [hunk 2]
```diff
@@ -73,36 +70,42 @@ function md5(word) {
...
  *
  * @param {Object} settings
  * @param {Object} options
- * @return {Objsect} completed `settings` object
+ * @return {Objsect} merged `settings` object
  * @api private
  */
-
-function complete (settings, options) {
+function merge (settings, options) {
```
Comment (cristiandouce) [inline index.js:100]: Missing newline.

File: index.js [hunk 3]
```diff
@@ -31,26 +30,25 @@ module.exports = function plugin(schema, options) {
  * @return {String} gravatar's API image `url`
  * @api private
  */
-
-function gravatar(settings) {
+function gravatar (settings) {
```
Comment (cristiandouce) [inline index.js:36]: Missing newline here too. I prefer the separation.

File: index.js [hunk 4]
```diff
@@ -13,14 +13,13 @@ var keys = Object.keys;
  * @param {Object} options
  * @api public
  */
-
-module.exports = function plugin(schema, options) {
+module.exports = function plugin (schema, options) {
```
Comment (cristiandouce) [inline index.js:16]: Again in here.

**Comment by cristiandouce** [comment]
I'd like to merge this. Any chance to update?

Again, thanks for your contributions! Great stuff!

**Comment by nijikokun** [comment]
You don't put newlines after docblocks that describe something, otherwise not all generators output correct details when doing code analysis, html generation, etc... They should be touching the code they are describing.

I have seen popular libraries do this however; but it's not to spec.

**Comment by cristiandouce** [comment]
I do! :dancers:

**Comment by nijikokun** [comment]
This means that certain parsers won't properly recognize the blocks and output incorrect information.

**Comment by cristiandouce** [comment]
I prefer code readability than comply to a parser that requires your comment block to _touch_ the code...

Mind rebase?

**Comment by nijikokun** [comment]
I'll rebase, but I'm afraid I don't agree with that sentiment, so you'll have to do that. Functionality holds higher in my eyes than style or opinion of readability.

**Comment by cristiandouce** [comment]
:+1: 

Did you close this on purpose?

**Comment by nijikokun** [comment]
Yes, to avoid the commit clutter

**Pull request closed without merging**